NSSLServer.Plugin.Recipes/Controller/RecipeController.cs

```diff
@@ -0,0 +1,309 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using NSSLServer.Database;
+using NSSLServer.Database.Attributes;
+using NSSLServer.Plugin.Shoppinglist.Manager;
+using System.Net.Http.Json;
+using System.Text.RegularExpressions;
+using static NSSLServer.Shared.RequestClasses;
+
+namespace NSSLServer.Plugin.Recipes.Controller
+{
+    [Route("recipe"), WithDbContext]
+    public class RecipeController : AuthenticatingDbContextController
+    {
+        private const string BaseUrl = "https://api.chefkoch.de/v2/recipes/";
+
+        private static readonly Regex IdRegex = new(@"(\d{10,})", RegexOptions.IgnoreCase);
+
+        private readonly HttpClient _httpClient;
+
+        public RecipeController(HttpClient httpClient)
+        {
+            _httpClient = httpClient;
+        }
+
+        [HttpPost, Route(nameof(CreateShoppingListForRecipe))]
+        public async Task<IActionResult> CreateShoppingListForRecipe([FromBody] AddRecipeArgs urlOrId, string? optionalListName = null, int? amountOfPeople = 4)
+        {
+            var recipe = await DownloadRecipe(urlOrId.IdOrUrl);
+
+            if (recipe is null)
+                return NotFound(urlOrId.IdOrUrl);
+
+            var recipeName = recipe.Title;
+
+            if (!string.IsNullOrWhiteSpace(optionalListName))
+                recipeName = optionalListName;
+            else if (string.IsNullOrWhiteSpace(recipeName))
+                recipeName = $"Neues Rezept {DateTime.Now.ToShortDateString()}";
+
+            var list = await ShoppingListManager.AddList(Context, recipeName, Session.Id);
+            foreach (var item in recipe.IngredientGroups.SelectMany(x => x.Ingredients))
+            {
+                if (item is null)
+                    continue;
+
+                var amount = item.Amount == 0 ? "" : item.Amount.ToString();
+
+                await ShoppingListManager.AddProduct(Context, list.Id, Session.Id,
+                    $"{item.Name} {amount} {item.Unit}{item.UsageInfo}".Trim(), null, 1, null);
+            }
+
+            return Json(await ShoppingListManager.LoadShoppingList(list.Id, false, Session.Id));
+        }
+
+        [HttpGet, Route(nameof(GetRecipe))]
+        public async Task<IActionResult> GetRecipe(string urlOrId)
+        {
+            var recipe = await DownloadRecipe(urlOrId);
+
+            return recipe is null ? NotFound(urlOrId) : Json(recipe);
+        }
+
+        private async Task<Recipes?> DownloadRecipe(string urlOrId)
+        {
+            var match = IdRegex.Match(urlOrId);
+
+            if (!match.Success)
+                return default;
+
+            var res = await _httpClient.GetFromJsonAsync<Recipes>($"{BaseUrl}{match.Value}");
+
+            return res;
+        }
+    }
+
+    public class Recipes
```

Comment:
Maybe you could remove the properties which aren't required for the functionality or atleast move it into another class